File: SpringBootApp/src/main/java/com/example/springbootapp/Data/DTO/ProductWithVariantDto.java
package com.example.springbootapp.Data.DTO;

import com.example.springbootapp.Data.Entities.Enums.Size;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class ProductWithVariantDto {
    private String id;
    private Size size;
    private BasicProductDto product;
}

File: ClientUser/app/src/main/java/com/example/clientuser/model/dto/ProductWithVariantDto.kt
package com.example.clientuser.model.dto

import com.example.clientuser.model.dto.ProductDto
import com.example.clientuser.model.enumerator.Size
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ProductWithVariantDto(
    val id: String,
    val size: Size,
    val product: BasicProductDto
)
